storage/ndb/rest-server/data-access-rondb/src/db-operations/pk/common.cpp

```diff
@@ -932,3 +937,47 @@ int GetByteArray(const NdbRecAttr *attr, const char **first_byte, Uint32 *bytes)
     return -1;
   }
 }
+
+bool CanRetryOperation(RS_Status status) {
+  bool retry = false;
+  if (status.http_code != SUCCESS) {
+    if (status.classification == NdbError::TemporaryError) {
+      retry = true;
+    } else if (UnloadSchema(status)) {
+      retry = true;
+    }
+  }
+
+  if (retry) {
+    DEBUG(std::string("Transient error. MySQL Code: ") + std::to_string(status.mysql_code) +
+          " Code: " + std::to_string(status.code));
+  }
+  return retry;
+}
+
+bool UnloadSchema(RS_Status status) {
+  bool unload = false;
+  if (status.http_code != SUCCESS) {
+    if (/*Invalid schema object version*/
+        (status.mysql_code == HA_ERR_TABLE_DEF_CHANGED && status.code == 241)) {
+      unload = true;
+    } else if (/*Table is being dropped*/
+               (status.mysql_code == HA_ERR_NO_SUCH_TABLE && status.code == 283)) {
+      unload = true;
+    } else if (/*Table not defined in transaction coordinator*/
+               (status.mysql_code == HA_ERR_TABLE_DEF_CHANGED && status.code == 284)) {
+      unload = true;
+    } else if (/*No such table existed*/
+               (status.mysql_code == HA_ERR_NO_SUCH_TABLE && status.code == 709)) {
+      unload = true;
+    } else if (/*No such table existed*/
+               (status.mysql_code == HA_ERR_NO_SUCH_TABLE && status.code == 723)) {
+      unload = true;
+    } else if (/*Table is being dropped*/
+               (status.mysql_code == HA_ERR_NO_SUCH_TABLE && status.code == 1226)) {
```

Comment:
Why even check the status.code? `HA_ERR_NO_SUCH_TABLE` should suffice? Or at least add a comment why it doesn't.